### ISSUE TITLE
Use of `eval` in development mode violates Content-Security-Policy

### ISSUE THREAD

**Issue by unindented** [ReactTraining/react-router#6611]
## Version

`4.4.0-beta.7`

## Test Case

*CodeSandbox doesn't support adding [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP), but it would look like this:*

```html
<html>
<head>
  <meta http-equiv="Content-Security-Policy" content="default-src 'none'; script-src 'self'; style-src 'self' 'unsafe-inline';" />
</head>
<body>
  <div id="root"></div>
  <script src="index.js"></script>
</body>
</html>
```

## Steps to reproduce

Import `react-router` in some file, launch app in development mode.

## Expected Behavior

Things loading fine with CSP enabled.

## Actual Behavior

Browser blocks execution of `react-router.js` because it uses `eval`:

<img width="404" alt="screen shot 2019-03-03 at 4 57 22 am" src="https://user-images.githubusercontent.com/69485/53695486-230a7f00-3d71-11e9-9178-00a3bcb6c355.png">

<img width="1439" alt="screen shot 2019-03-03 at 5 06 36 am" src="https://user-images.githubusercontent.com/69485/53695582-26523a80-3d72-11e9-8705-da25ad09a19f.png">

Is there a way of implementing your current logic without `eval`?

**Comment by unindented** [comment]
(This issue wasn't present in `4.4.0-beta.6` btw.)

**Comment by emmenko** [comment]
Having the same issue, I was about to report it as well. It was introduced here (https://github.com/ReactTraining/react-router/blob/b2c6fa0725b7ff1ed762064d633b26b6293e0140/packages/react-router/modules/index.js#L1-L20)

@mjackson what was the reason for using `eval`?

**Comment by mjackson** [comment]
Thanks for the report, @unindented. The purpose of [using `eval` indirectly](http://www.ecma-international.org/ecma-262/5.1/#sec-10.4.2) here is to get a reference to the global object. I'll try another method and cut a new beta release.

**Comment by mjackson** [comment]
BTW, does anyone know what CSP Codesandbox actually uses?

**Comment by emmenko** [comment]
Thanks for fixing that! 💯

One small thing to clarify: it’s not just about codesandbox in particular but any app that uses a CSP in development mode.

Anyway, thanks again for looking into that 🤗

**Comment by mjackson** [comment]
The fix for this was just [released in 4.4.0-beta.8](https://github.com/ReactTraining/react-router/releases/tag/v4.4.0-beta.8). Please give it a shot.

@emmenko You can still use CSP with `eval`, you just have to include `'unsafe-eval'`

**Comment by unindented** [comment]
@mjackson verified on my app, this is working perfectly. Thank you!
